packages/language-server/index.js

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env node
```

Comment:
Very few comments here, would be useful to explain what does what

Reply:
It’s the core file that integrates all user events with certain actions in utilities.
I think this is the place where you can explain what user actions will result in what “computer responses”.
For example, in the `connection.onHover` handler, you can explain what a user has
to do to trigger this event, and what we’re going to do in response.
This might seem really clear (well, user has to hover, something will be shown in response), but, reading that code, I can see that you call some utility, but I don’t know what you do with that, why does it always return TypeScript that can be wrapped in a fenced code block ` ```ts `?
Or, reading `connection.onPrepareRename`, what will it do? What is the difference between prepare and request?

Reply:
Some of these are just copy-pasted from other implementations. I.e. I looked at some TypeScript language servers I found, the official TypeScript integration in VSCode, the Monaco editor TypeScript integration. I don’t know why it’s formatted like that, but it works and it’s consistently used like that in the ecosystem.

I don’t really feel the need to document each request type, because it’s already explained in the [LSP specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification).